crypto/tss/recovery/recover_private_key.go

```diff
@@ -0,0 +1,73 @@
+package recovery
+
+import (
+	"crypto/ecdsa"
+	"errors"
+	"fmt"
+	"math/big"
+
+	"github.com/getamis/alice/crypto/birkhoffinterpolation"
+	"github.com/getamis/alice/crypto/ecpointgrouplaw"
+	"github.com/getamis/alice/crypto/elliptic"
+	"github.com/getamis/alice/crypto/utils"
+)
+
+var (
+	ErrNotEnoughPeers = errors.New("not enough input peers, need at least 2")
+	ErrAbsentCurve    = errors.New("curve is nil")
+	ErrPubKeyMismatch = errors.New("pubkey derved from recovered privkey is not equal to pubkey provided")
+)
+
+type RecoveryPeer struct {
+	share *big.Int
+	bk    *birkhoffinterpolation.BkParameter
+}
+
+func RecoverPrivateKey(curve elliptic.Curve, threshold uint32, pubKey *ecpointgrouplaw.ECPoint, peers []RecoveryPeer) (*ecdsa.PrivateKey, error) {
```

Comment:
A point to consider here: i chose to describe function arguments in `NewDkg/NewSigner` arguments style to keep it simple and unambiguous. But i suppose actual use case of such function may be that library user provides array of DKG results. I have higher-level function that accepts such array and does necessary validation (BKs/Pubkey/Threshold consistency in each array entry for example to check its same peer quorum results) but not sure if it makes sense to keep that logic as part of Alice. Altough i'll be happy to add such function here if it makes sense.

Reply:
I think it's acceptable for this version. Your consideration is meaningful.